Process stale globalIPs present on non-Gateway nodes

sridhargaddam opened the pull request:
In certain deployments, due to timing issues, globalIP annotation on non-Gateway nodes were not removed. 
As a result, there's a chance that globalIPs for exported services might unintentionally be updated with new globalIPs when such nodes become active gateway nodes. This PR fixes this issue and also improves logging in some of the globalnet controllers.

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

sridhargaddam committed:
Improve logging in Globalnet controllers

Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>

tpantelis commented on pkg/globalnet/controllers/base_controllers.go:
```diff
@@ -154,6 +154,8 @@ func (c *baseIPAllocationController) reserveAllocatedIPs(federator federate.Fede
 		return federator.Distribute(obj) //nolint:wrapcheck  // Let the caller wrap it
 	}
 
+	logger.Infof("Successfully reserved globalIP of %q for %s/%s", reservedIPs, obj.GetNamespace(), obj.GetName())
```
```suggestion
	logger.Infof("Successfully reserved GlobalIPs %q for %s \"%s/%s\"", reservedIPs, obj.GetKind(), obj.GetNamespace(), obj.GetName())
```

tpantelis commented on pkg/globalnet/controllers/node_controller.go:
```diff
@@ -99,10 +99,13 @@ func (n *nodeController) process(from runtime.Object, _ int, op syncer.Operation
 	if node.Name != n.nodeName {
```
How about:

`logger.Infof("Processing %sd non-gateway node %q - releasing GlobalIP %q", op, node.Name, existingGlobalIP)`

instead of the 2 below.

Also on L189 in `reserveAllocatedIP` below, can you add:

`logger.Infof("Successfully reserved allocated GlobalIP %q for node %q", existingGlobalIP, obj.GetName())
`

tpantelis reviewed (changes requested):
In _gateway_monitor.go_, L145 in `handleCreatedOrUpdatedEndpoint`, change:;

`logger.V(log.DEBUG).Infof("In processNextEndpoint, endpoint info: %+v", endpoint)`

to

`logger.V(log.DEBUG).Infof("Gateway monitor informed of create/updated endpoint: %v", endpoint)`

Also similarly,  L231:

`logger.V(log.DEBUG).Infof("Informed of removed endpoint for gateway monitor: %v", endpoint)`

to

`logger.V(log.DEBUG).Infof("Gateway monitor informed of removed endpoint: %v", endpoint)`

sridhargaddam committed:
Process any stale globalIPs present on non-Gateway nodes

In certain deployments, due to timing issues, globalIP annotation
on non-Gateway nodes were not removed. As a result, there's a chance
that globalIPs for exported services might unintentionally be updated
with new globalIPs when such nodes become active gateway nodes.

Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>

skitt committed:
Merge branch 'devel' into improve-gn-logging

sridhargaddam commented:
@skitt @dfarrell07 the OVN jobs are failing with the following error.

```
/go/pkg/mod/golang.org/x/net@v0.14.0/html/charset/charset.go:20:2: unzip 
/go/pkg/mod/cache/download/golang.org/x/text/@v/v0.12.0.zip: write 
/go/pkg/mod/golang.org/x/text@v0.12.0/unicode/norm/data15.0.0_test.go: no space left on device

FAIL	github.com/submariner-io/submariner/test/e2e [setup failed]
```
